### DISCUSSION TITLE
How to paste from the clipboard into the terminal

### DISCUSSION THREAD

**Discussion by sidouglas** [junegunn/fzf#3952] [Q&A] [answered]
Similar Q: https://github.com/junegunn/fzf/discussions/3081

I apologise for the basic question, but all I want to perform is to paste contents of my clipboard into the terminal without fzf being triggered.
* fzf --version `0.54.3 (brew)`
* zsh on M2 mac 14.5 (23F79)
* zsh 5.9 (arm-apple-darwin22.1.0)
config:
```
echo 'fzf loaded';

# Set up fzf key bindings and fuzzy completion
source <(fzf --zsh)

# Auto-completion
# ---------------
[[ $- == *i* ]] && source "/opt/homebrew/opt/fzf/shell/completion.zsh" 2> /dev/null
export FZF_DEFAULT_OPTS='--multi --no-height --extended'
export FZF_DEFAULT_COMMAND='rg --hidden --files -g "!.git" -g "!node_modules/*" -g "node_modules"'
# CTRL-Y to copy the command into clipboard using pbcopy
export FZF_CTRL_R_OPTS="
  --bind 'ctrl-y:execute-silent(echo -n {2..} | pbcopy)+abort'
  --color header:italic
  --header 'Press CTRL-Y to copy command into clipboard'"
```

https://github.com/user-attachments/assets/a9ea5c27-d084-4d98-a28f-269e4db7c6e8
* I have to hit esc (⎋ twice) in order to cancel fzf and perform a simple paste from clipboard.

**Comment by LangLangBart** [comment, marked as answer]
> * I have to hit esc (⎋ twice) in order to cancel fzf and perform a simple paste from clipboard.

I can't reproduce it so far. When you run the following commands, is the issue still observable?

```sh
# minimal zsh setup
command env -i HOME=$HOME TERM=$TERM USER=$USER PATH=$PATH SHELL=zsh zsh --no-rcs

# fzf setup
source <(fzf --zsh)
export FZF_DEFAULT_OPTS='--multi --no-height --extended'
export FZF_CTRL_R_OPTS="--bind 'ctrl-y:execute-silent(echo -n {2..} | pbcopy)+abort'"

# Press ctrl-r
# Press ctrl-y to copy a command
# Press Esc once
```

If the issue is no longer observable, it is very likely an issue with your shell setup, likely a plugin[^1]. Try to disable some plugins/options until you find the one causing the mischief.

[^1]: [dotfiles/zshrc · sidouglas/dotfiles · GitHub](https://github.com/sidouglas/dotfiles/blob/1beefef8924631d94b8f761a3c850e6c207411f7/zshrc?plain=1#L81)

**Reply by sidouglas** [reply]
Thank you. It was my alias `fc` — that's reserved. In the end I used another alias.

**Reply by LangLangBart** [reply]
If it's working correctly now, that's good, but the answer is unexpected. The expansion of aliases should be fixed with the `no_aliases` option set in the `fzf-history-widget`, and parsing should also disable alias expansion when sourcing the function since #1944 and #3794.

https://github.com/junegunn/fzf/blob/4e85f72f0ee237bef7a1617e0cf8c811a4091d72/shell/key-bindings.zsh#L107-L111
